### PR TITLE
Support App Bridge 2.0 in shopify_app

### PR DESCRIPTION
**Pull request by rezaansyed**
### What this PR does

> ⚠️ This is a breaking change

Support App Bridge 2.0 in shopify_app.

### Reviewer's guide to testing

- Import local version of this branch to new app
- Ensure that app runs as expected in Admin

### Things to focus on

Ensure migration strategy is followed as per the [guide](https://shopify.dev/tutorials/migrate-your-app-to-app-bridge-2).

### Checklist

Before submitting the PR, please consider if any of the following are needed:

- [x] Update `CHANGELOG.md` if the changes would impact users
- [x] Update `README.md`, if appropriate.
- [x] Update any relevant pages in `/docs`, if necessary
- [x] For security fixes, the [Disclosure Policy](https://github.com/Shopify/shopify_app/blob/master/SECURITY.md#disclosure-policy) must be followed.

### CODE REVIEW

**Review by hannachen — APPROVED** [review]
Looks good to me 👍

**Review by paulomarg — CHANGES_REQUESTED** [review]
I've tried this out now, and I got the following error:
```
message: "APP::ERROR::INVALID_CONFIG: host must be provided"
name: "AppBridgeError"
type: "APP::ERROR::INVALID_CONFIG"
```

It seems that coming back from OAuth, we're lacking the `host` parameter and thus we can't load App Bridge properly.

**Comment by rezaansyed** [comment]
> I've tried this out now, and I got the following error:
> 
> ```
> message: "APP::ERROR::INVALID_CONFIG: host must be provided"
> name: "AppBridgeError"
> type: "APP::ERROR::INVALID_CONFIG"
> ```
> 
> It seems that coming back from OAuth, we're lacking the `host` parameter and thus we can't load App Bridge properly.

Noticing that on my end too. It has to do with the `HomeController` not having the `host` param passed in and can't initialize App Bridge at the top level. Is there any way for us to know of the host during OAuth @hannachen?

**Comment by NabeelAhsen** [comment]
If the `host` parameter is part of the callback URL sent by Shopify to the app, step 3 of this doc might be outdated: https://shopify.dev/tutorials/authenticate-with-oauth#step-3-confirm-installation

Is this where the `host` parameter is initially supposed to come from?

**Comment by rezaansyed** [comment]
> If the `host` parameter is part of the callback URL sent by Shopify to the app, step 3 of this doc might be outdated: https://shopify.dev/tutorials/authenticate-with-oauth#step-3-confirm-installation
> 
> Is this where the `host` parameter is initially supposed to come from?

It's not part of the callback controller. The failure happens when hitting the home controller at the top level right after the callback controller though.

**Comment by paulomarg** [comment]
Should we be including the host in the redirect URL when we begin OAuth?

File: lib/shopify_app/controller_concerns/login_protection.rb [hunk 1]
```diff
@@ -216,8 +224,8 @@ def sanitized_params
 
     def return_address
       return base_return_address unless ShopifyApp.configuration.allow_jwt_authentication
-      return_address_with_params(shop: current_shopify_domain)
-    rescue ShopifyDomainNotFound
+      return_address_with_params(shop: current_shopify_domain, host: host)
+    rescue ShopifyDomainNotFound, ShopifyHostNotFound
```
[the remark this answers is not in the document]
Comment (NabeelAhsen) [reply lib/shopify_app/controller_concerns/login_protection.rb:228]: Things I'm unsure about here:
- Right now, there are two known places that use `return_address` in this code base. Is raising when `host` isn't found the the correct move here?
- What other tests are affected by this?

**Comment by NabeelAhsen** [comment]
> Should we be including the host in the redirect URL when we begin OAuth? That way, we'd be able to reference it when coming back.

This `host` param comes from Shopify's generated URLs, so we'd only be aware of them on OAuth return

File: lib/shopify_app/controller_concerns/login_protection.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (paulomarg) [reply lib/shopify_app/controller_concerns/login_protection.rb:228]: Considering we currently throw a similar exception for the domain, it makes sense to me to continue to throw here.

**Review by paulomarg — APPROVED** [review]
🎩 ed again, OAuth worked correctly for me now.

**Comment by asecondwill** [comment]
This does not work when trying to update a JWT Turbo Example app.  Please can you test with that and provide instructions @NabeelAhsen

**Comment by asecondwill** [comment]
Probably worth noting,  host is not passed back in the params @paulomarg

**Pull request merged**